### DISCUSSION TITLE
Unable to create English Auction

### DISCUSSION THREAD

**Discussion by mohanv-inmar** [ProjectOpenSea/opensea-js#1159] [Q&A] [answered]
Hey, 

I'm trying to create English Auction programatically, as shown in the [docs here](https://github.com/ProjectOpenSea/opensea-js#creating-english-auctions)

I'm adding `waitForHighestBid: true` when passing the object to `createSellOrder` as shown in the docs.
but I'm unable to do so, as I'm using openseajs sdk with typescript and when I check the types for the `createSellOrder` function it doesn't have `waitForHighestBid` field. (attaching the image for reference)
<img width="736" alt="image" src="https://github.com/ProjectOpenSea/opensea-js/assets/138445993/41459c2c-ac1b-4520-83cf-857d7b6d56f6">

I'm doing `ts-ignore` to set the field anyway, still the API is not working as expected. It is creating a normal listing instead of creating a english auction. 
<img width="590" alt="image" src="https://github.com/ProjectOpenSea/opensea-js/assets/138445993/d8ba88f2-1ce1-4262-b8e4-9cb4248cbff9">

**Comment by ryanio** [comment, marked as answer]
Thanks for pointing out, seems like we were missing the feature completely, have opened a PR to fix (you'll use `englishAuction: true` in createSellOrder) https://github.com/ProjectOpenSea/opensea-js/pull/1160

**Reply by mohanv-inmar** [reply]
thanks @ryanio

**Comment by mohanv-inmar** [comment]
I'm using `mumbai` chain in opensea.
As the SDK says, `only WETH is currently supported for chains other than Mainnet Ethereum`

I'm trying to use polygon mumbai WETH token address `0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa` as `paymentTokenAddress` while doing `createSellOrder`.  You can check the token details [on polygonscan here](https://mumbai.polygonscan.com/token/0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa)

```ts
    const paymentTokenAddress = '0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa';

    return this.openseaSdk.createSellOrder({
      asset: {
        tokenId: tokenId,
        tokenAddress: tokenAddress,
      },
      accountAddress: offerer,
      startAmount: baseAmount,
      listingTime: parseInt(startTime),
      expirationTime: parseInt(expirationTime),
      paymentTokenAddress,
      englishAuction: true,
    });
```

Even when I'm using WETH token address, It's throwing:
`Error: No ERC-20 token found for 0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa, only WETH is currently supported for chains other than Mainnet Ethereum`


The error is being thrown here in `sdk.ts`:
https://github.com/ProjectOpenSea/opensea-js/blob/8150be60f982909eec45dbf5e509ab4c019fc194/src/sdk.ts#L1042-L1057

looks like the function `getWETHAddress` is throwing error as the WETH for `mumbai` chain is not added there.
https://github.com/ProjectOpenSea/opensea-js/blob/8150be60f982909eec45dbf5e509ab4c019fc194/src/utils/utils.ts#L324-L357


Can you please add Mumbai chain WETH address to `getWETHAddress` function ? Or Is there some other way I can solve this issue ?

**Reply by mohanv-inmar** [reply]
@ryanio  I've also raised a PR for the same, #1162
